### DISCUSSION TITLE
Is there a schema that describes the goreleaser config format?

### DISCUSSION THREAD

**Discussion by myitcv** [goreleaser/goreleaser#2065] [Q&A] [answered]
Hi - I've had a look through issues and discussions, and had a general Google search but I can't find anything that describes the schema of the goreleaser config, much like https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/github-workflow.json describes the schema for GitHub workflow definitions. 

Does such a thing exist?

Many thanks

**Comment by caarlos0** [comment]
Hey, we don't have that no. It would be nice to, though. 

I remember looking into that a while ago, but didn't find a good way to generate it from Go code...

**Reply by myitcv** [reply]
Thanks.

Can you point me at the relevant code here? Because we could well use CUE (https://cuelang.org/) to generate schemas directly from the Go code.

**Reply by caarlos0** [reply]
Sure thing, https://github.com/goreleaser/goreleaser/blob/master/pkg/config/config.go 

Been wanting to try CUE for a while as well :D

**Reply by myitcv** [reply]
Ok, just tried. This should be eminently doable with CUE, but unfortunately this example runs into two known issues:

* https://github.com/cuelang/cue/issues/668
* https://github.com/cuelang/cue/issues/648

Both are on my radar to fix. I'll add a note in those issues to circle back here.

**Reply by caarlos0** [reply]
noice, thanks @myitcv 🙏

**Comment by caarlos0** [comment, marked as answer]
FWIW, we do have JSONSchemas now: https://goreleaser.com/customization/
